### ISSUE TITLE
retry inside codecept.conf.js not working

### ISSUE THREAD

**Issue by shineqwe** [Codeception/CodeceptJS#3663]
#### I'm trying to set up a global option to re-run failed tests as written
https://codecept.io/basics/#retry-configuration

```js
export const config: CodeceptJS.MainConfig = {
   tests: 'tests/site/**/*_test.ts',
   output: './output',
   retry: 2
```
When it crashes, tests are not re-executed
framework version "codeceptjs": "^3.4.1"

**Comment by jhalborg** [comment]
Confirmed, we are experiencing the same issue. We also tried the more explicit retry configs and different combinations

```
retry: [
    {
      Scenario: 1,
      Feature: 1,
    },
    {
      grep: '@flaky',
      Scenario: 3,
      Feature: 2,
    },
    // Typings will be fixed in future version, PR already landed but is not published yet:
    // https://github.com/codeceptjs/CodeceptJS/pull/3602
    // Should be fixed once 3.4.2+ or 3.5 is published
  ]
  ```

but no retries are ever performed - also 3.4.1

**Comment by jhalborg** [comment]
Linking these two issues together: https://github.com/codeceptjs/CodeceptJS/issues/3607
